### ISSUE TITLE
linear combination of losses

### ISSUE THREAD

**Issue by Fan-Luo** [allenai/longformer#143]
Hi, 

For the loss computation of hotpotqa, you mentioned
> We combine span, question classification, sentence, and paragraphs losses and train the model in a multitask way using linear combination of losses.

With `linear combination of losses`, do you mean multiplying a weight hyper-parameter to each loss? would the model learn to assign all the weights to be 0 to minimize the loss? Or did you set the weights to be constant? 

Besides, for the span extraction loss, call the `or_softmax_cross_entropy_loss_one_doc` function as trivialqa?
for question type (yes/no/span) loss, call `torch.nn.CrossEntropyLoss()`
for supporting sentences and paragraphs loss, call `torch.nn.BCELoss(reduction='mean')`
Is this correct?

Thank you

**Comment by armancohan** [comment]
> With linear combination of losses, do you mean multiplying a weight hyper-parameter to each loss? 

yes, combine the losses linearly with assigning a constant weight to each loss. concretely:
`total_loss = lambda1 * span_loss + lambda2 * sentence_loss + lambda3 * paragraph_loss + lambda4 * type_loss`.
We set lambdas so that losses are in similar range. In our experience, tuning this doesn't impact results that much.

> Besides, for the span extraction loss, call the or_softmax_cross_entropy_loss_one_doc function as triviaqa?

Yes, we use the exact loss as triviaqa for span. All other losses are CrossEntropyLoss.

**Comment by Fan-Luo** [comment]
To use CrossEntropyLoss to compute supporting sentences and paragraphs loss. 

With the output from the two layer FF,
```
sp_para_output_t = self.fnn_sp_para(sequence_output[:,para_indexes,:])        # logit score for 'yes'
sp_sent_output_t = self.fnn_sp_sent(sequence_output[:,sent_indexes,:])
```
Argument the sp_para_output_t and sp_sent_output_t scores with additional 0 score of same size 
```
sp_para_output_aux = torch.zeros(sp_para_output_t.shape)                      # 0 score for 'no'
predict_support_para = torch.cat([sp_para_output_aux, sp_para_output_t], dim=-1)
sp_sent_output_aux = torch.zeros(sp_sent_output_t.shape)
predict_support_sent = torch.cat([sp_sent_output_aux, sp_sent_output_t], dim=-1)
```
Compute the loss:
```
sp_para_loss = torch.nn.CrossEntropyLoss(predict_support_para, gold_support_para_index)
sp_sent_loss = torch.nn.CrossEntropyLoss(predict_support_sent, gold_support_sent_index)
```
With the default `reduction = 'mean'` for CrossEntropyLoss, it is average loss over batch by definition. However, you use batch_size=1, and the average here is over the total number of  paragraphs / sentences. So that the final paragraph loss would be a average loss of the 10 paragraphs' losses, and the final sentence loss would be a average loss of all the sentences' losses. 
Is this all correct?

For question type loss, no need to argument the logits with 0s before calling CrossEntropyLoss, because its output layer has 4 nodes corresponding to span/yes/no/null, am I right?

Thanks a lot.
